src/nastran/nastran.cpp

```diff
@@ -0,0 +1,390 @@
+}
+
+void parseTRIAToIDList(const std::string& line, std::vector<int>& idArray) 
+{
+    std::regex regex(",");
+
+    std::vector<std::string> out(
+        std::sregex_token_iterator(line.begin(), line.end(), regex, -1),
+        std::sregex_token_iterator()
+    );
+
+    if (debugOutput) std::cout << "TRIA: ";
+
+    int index = 0;
+    for (auto& s : out) {
+        if (index <= 5 && index > 2) {
+            int id = std::stoi(s);
+            idArray.push_back(id);
+            if (debugOutput) std::cout << id << " ";
+        }
+        index++;
+    }
+    if (debugOutput) std::cout << std::endl;
+}
+
+void parseQUADToIDList(const std::string& line, std::vector<int>& idArray) 
+{
+    std::regex regex(",");
+
+    std::vector<int> unfoldIndices{0,1,2,2,3,0};
+    std::vector<int> quadIndices;
+
+    std::vector<std::string> out(
+        std::sregex_token_iterator(line.begin(), line.end(), regex, -1),
+        std::sregex_token_iterator()
+    );
+
+    if (debugOutput) std::cout << "QUAD: ";
+
+    int index = 0;
+    for (auto& s : out) {
+        if (index <= 6 && index > 2) {
+            int id = std::stoi(s);
+            quadIndices.push_back(id); 
+            if (debugOutput) std::cout << id << " ";
+        }
+        index++;
+    }
+
+    if (debugOutput) std::cout << " -> ";
+
+    for (auto unfoldIndex : unfoldIndices)
+    {
+        int unfoldedID = quadIndices[unfoldIndex];
+        idArray.push_back(unfoldedID);
+        if (debugOutput) std::cout << unfoldedID << " ";
+    }
+    if (debugOutput) std::cout << std::endl;
+}
+
+void parseTempToIDandFloatVec(const std::string& line, std::vector<float>& temperatureList, std::vector<int>& idList) 
+{
+    std::regex regex(",");
+
+    std::vector<std::string> out(
+        std::sregex_token_iterator(line.begin(), line.end(), regex, -1),
+        std::sregex_token_iterator()
+    );
+
+    int id = 0;
+    float temperature = 0.f;
+
+    int index = 0;
+    for (auto& s : out) {
+        switch (index)
+        {
+
+        case 2: 
+            id = std::stoi(s);
+            idList.push_back(id);
+            break;
+        
+        case 3: 
+            temperature = std::stof(s);
+            temperatureList.push_back(temperature);
+            break;
+        
+
+        }
+        index++;
+    }
+
+    if (debugOutput) std::cout << "Temp: [" << id << "] " << temperature << std::endl;
+}
+
+nastran::nastran() 
+{
+    debugOutput = enableDebugOutput;
+}
+
+vsg::ref_ptr<vsg::Object> nastran::read(const vsg::Path& filename, vsg::ref_ptr<const vsg::Options> options) const
+{
+    auto ext = vsg::lowerCaseFileExtension(filename);
+    if (ext != "nas" || !vsg::fileExists(filename)) return { };
+    
+
+    std::ifstream stream(filename.c_str(), std::ios::in);
+    if (!stream) return { };
+    return read(stream, options);
+
+    return vsg::ref_ptr<vsg::Object>();
```

Comment:
I presume this line of code is hang up of dev process/debugging.  This will need to be removed.